### ISSUE TITLE
False positive errors when linting specific sequence of HTML strings

### ISSUE THREAD

**Issue by outoftime** [htmllint/htmllint#213]
I’m not sure what it is about these strings, but if you lint:

```html
<!DOCTYPE html>
<html>
    <head>
        <title>Page Title</title>
    </head>
    <body>
        <input id="someid" placeholder="Input>
    </body>
</html>
```

Followed by:

```html
<!DOCTYPE html>
<html>
    <head>
        <title>Page Title</title>
    </head>
    <body>
        <input id="someid" placeholder="Input">
    </body>
</html>
```

`htmllint` will report the second HTML string (which is in fact valid) as having a duplicated id of `someid` as well as a missing attribute value on the `<html>` tag.

Here’s a full reduction:

```js
const htmllint = require('htmllint')

const s1 = `<!DOCTYPE html>
<html>
    <head>
        <title>Page Title</title>
    </head>
    <body>
        <input id="someid" placeholder="Input>
    </body>
</html>
`;

const s2 = `<!DOCTYPE html>
<html>
    <head>
        <title>Page Title</title>
    </head>
    <body>
        <input id="someid" placeholder="Input">
    </body>
</html>
`;

async function test() {
  await htmllint(s1);
  console.log(await htmllint(s2));
}

test();
```

**Comment by outoftime** [comment]
BTW, this seems to be some kind of state getting incorrectly retained in the `defaultLinter`. Instantiating a new `Linter` each time resolves the issue.

**Comment by yvanavermaet** [comment]
@outoftime Was able to reproduce and it seems it's an issue with the first string you provide. The value of the placeholder attribute is missing a double quote at the end.

**Comment by mlochbaum** [comment]
Evidently `end` from lib/rules/id-no-dup.js isn't being called. That's pretty weird, since the function that should do that (resetRules) is wrapped in a finally block in the linter.

Thanks for providing a good repro, though. Shouldn't be too hard to debug...
